tripleo-quickstart-config/roles/libvirt/setup/overcloud/tasks/main.yml

```diff
@@ -96,8 +96,9 @@
 # Generate the ironic node inventory files.  Note that this
 # task *must* occur after the above overcloud tasks, because if
 # `overcloud_nodes` is defined the template depends on the
-# `node_mac_map` variable.
+# `node_mac_map` variable. If the file exists - won't overwrite
 - name: Write ironic node json files
   template:
     src: ../templates/ironic_nodes.json.j2
     dest: "{{ working_dir }}/ironic_nodes.json"
+    force: no
```

Comment:
Hmm, we need to be sure this is overwritten in the VM case (I think it's cleaned on teardown, but we've got be be sure it's not stale).

I think it would be better to leave this alone, and disable quickstart completely in the baremetal case?

Then we can just write $WORKING_DIR/ironic_nodes.json in the baremetal case (and add a test to ensure it exists)?

So add a conditional for baremetal to either skip the ansible run (or perhaps use tags if we still need the libvirt setup for the bootstrap VM?) here:

https://github.com/metalkube/dev-scripts/blob/master/02_configure_host.sh#L13


Then add a test that this file exists here, and it can be manually created:

https://github.com/metalkube/dev-scripts/blob/master/05_deploy_bootstrap_vm.sh#L111

I'm happy to make these changes in a seperate PR if you can drop the force: no from this one?

Reply:
Also can you please drop the instack.json from the commit message (or if we leave something related to the nodes file here we need to say ironic_nodes.json as instack is a tripleo thing which we renamed here.

Reply:
Ah, this reminds me @hardys ironic_nodes.json gets red to generate the DNS configuration -- mapping the mac address to the hostname and IP. 

We should have `common.sh` have 
```
NODES=${NODES:-}
```
If someone is doing a BM deployment, they add the node.json here, if it is blank we run through quickstart -- use this as the conditional to determine if we run through creating the VMBC?

Reply:
Yeah it can either be a variable which accepts a json string, or a filename, doesn't really matter I guess, and yes then we can use it as a conditional to either skip quickstart completely or just run the bits we need to enable booting the bootstrap VM.